### ISSUE TITLE
Different Android Folder Structure using `react-native upgrade` vs `react-native init projectname`

### ISSUE THREAD

**Issue by idibidiart** [facebook/react-native#4107]
<img width="273" alt="screen shot 2015-11-12 at 5 06 13 pm" src="https://cloud.githubusercontent.com/assets/1288823/11136046/e330e4d2-895f-11e5-9d0c-99fe31c91f5e.png">

<img width="273" alt="screen shot 2015-11-12 at 5 06 05 pm" src="https://cloud.githubusercontent.com/assets/1288823/11136037/cd4d2a86-895f-11e5-9d62-68c367cabc03.png">

The editor window on top is the result of `react-native init projectName` while the one below is the result of running `react-native upgrade` on an existing 0.13.2 project (after changing react-native version to 0.14.2)

Is this how it's supposed to be or will we run into trouble given the variance in folder structure between an upgraded project and a new one?

**Comment by mkonicek** [comment]
cc @foghina

**Comment by mkonicek** [comment]
Can you capture the whole structure for both? To me it looks like both images are cut off and therefore I can't tell what's different. What's the actual difference?

**Comment by idibidiart** [comment]
This `android` folder was created with `react-native init`
 
<img width="632" alt="screen shot 2015-11-14 at 12 44 00 pm" src="https://cloud.githubusercontent.com/assets/1288823/11165745/867adf4e-8acd-11e5-90ba-2d8dd19aef77.png">


This `android` folder was created via `react-native upgrade`

<img width="512" alt="screen shot 2015-11-14 at 12 43 04 pm" src="https://cloud.githubusercontent.com/assets/1288823/11165747/a8d00380-8acd-11e5-9f7c-26a2aa3c3c4d.png">

Note the `images` folder is outside of the `android` folder so nothing is cut off here, you're seeing the entire contents of the `android` folder in both cases @mkonicek

**Comment by foghina** [comment]
@idibidiart So you're missing `res`? That folder is not regenerated when running `upgrade` as it contains mostly binaries that we don't want to diff and things that we generally don't change in the template. How did you run `upgrade`? Looks like your root folder is different in the two screenshots, and also your package name is different.

**Comment by idibidiart** [comment]
@foghina

They are two different projects one created with init (the one having the red folders) and the other created with upgrade. The upgrade was run from inside the project's root folder as: react-native upgrade. I expected the android folder in both projects to have an identical structure. It is not just missing the red folders in the case when it is generated via upgrade but it is also missing the  gradle folder and wrapper subfolder and the gradlew files (although I am on OS X so I can understand if those are not be generated when running upgrade) 

So basically, will I have a problem without the res and gradle/wrapper folders? 

Should I simply create a new project with init and bring my existing JS, ObjC and image files into it? That's what I ended up doing but it's a hassle. I want to make sure that upgrade is not resulting in missing files/folders that would cause android builds to fail. I have yet to start the android side of development, and just wanting to make sure the difference in folder structure and content for the android folder is as intended.

**Comment by foghina** [comment]
Yeah, the Android app won't work without those folders. `react-native upgrade` is only meant to be used in pre-existing React Native projects, it's not meant for adding React Native to a pre-existing node project that didn't have RN at all.

**Comment by idibidiart** [comment]
@foghina

No you are missing a bit of info: the pre-existing project had React 013.0 in it! It was working fine but I wanted to upgrade to 0.14.2 so I change the react-native version in package.json to 0.14.2 and ran react-native upgrade

What am *I* missing?

**Comment by foghina** [comment]
@idibidiart did this project start out as iOS-only (i.e. before we open-sourced Android)? If yes, then you should run `react-native android` first, in order to add the whole Android support to it. Then you don't even need `upgrade` until the next version.

**Comment by idibidiart** [comment]
Yes, I see, but that is not mentioned in the docs. I assume many folks started out with an iOS project since Android wasn't ready at start and running react-native android prior to reat-native upgrade should be spelled out somewhere as it could lead to confusion and further issues down the line.

http://facebook.github.io/react-native/docs/upgrading.html#content

**Comment by mkonicek** [comment]
Questions like this are best suited for Stack Overflow before filing an actual bug: http://stackoverflow.com/questions/tagged/react-native

I'll close this as Felix explained how the upgrade command works.

**Comment by mkonicek** [comment]
@idibidiart Can you send a PR making the docs clearer please?

**Comment by idibidiart** [comment]
Sure. I'll find some time this weekend!

For future reference, if the docs are missing important pieces of info or are out dated, shouldn't we file an issue here on the repo? Or are you suggesting we file issues only for software bugs?  @mkonicek
